File: leet_318_max_word_len_prod___bruteforce.cpp
class Solution {
public:
    int maxProduct(vector<string>& words)
    {
        int n = words.size();
        int res = 0, i, j;
        vector<vector<int>> count(n, vector<int>(26, 0));

        i = -1;
        while (++i < n)
        {
            for (char c: words[i])
                count[i][c - 'a']++;
            j = -1;
            while (++j < i)
                if (cmp(count[i], count[j]))
                    res = max((int) words[i].length() * (int) words[j].length(), res);
        }
        return res;
    }

    int cmp(vector<int>& a, vector<int>& b) // using & is faster
    {
        int i = -1;
        while (++i < 26)
            if (a[i] && b[i])
                return false;
        return true;
    }
};

File: leet_318_max_word_len_prod___bruteforce.py
class Solution:
    def maxProduct(self, words: List[str]) -> int:
        def cmp(a: List[int], b: List[int]) -> bool:
            for i in range(26):
                if a[i] > 0 and b[i] > 0:
                    return False
            return True
        res = 0
        n = len(words)
        count = [[0] * 26 for _ in range(n)]
        for i in range(n):
            for c in words[i]:
                count[i][ord(c) - 97] += 1
            for j in range(i):
                if cmp(count[i], count[j]):
                    res = max(len(words[i]) * len(words[j]), res)
        return res
